internal/data/accounts.go

```
@@ -34,14 +34,23 @@ func (m *AccountModel) Get(ctx context.Context, address string) (*types.Account,
 func (m *AccountModel) Insert(ctx context.Context, address string) error {
 	const query = `INSERT INTO accounts (stellar_address) VALUES ($1) ON CONFLICT DO NOTHING`
 	start := time.Now()
-	_, err := m.DB.ExecContext(ctx, query, address)
+	result, err := m.DB.ExecContext(ctx, query, address)
 	duration := time.Since(start).Seconds()
 	m.MetricsService.ObserveDBQueryDuration("INSERT", "accounts", duration)
 	if err != nil {
 		return fmt.Errorf("inserting address %s: %w", address, err)
 	}
 	m.MetricsService.IncDBQuery("INSERT", "accounts")
 
+	// Check if any rows were affected to determine if account already exists
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("checking rows affected for address %s: %w", address, err)
+	}
+	if rowsAffected == 0 {
+		return fmt.Errorf("account %s is already registered", address)
+	}
+
```

:wave: The query has `ON CONFLICT DO NOTHING`, so the idea when applying that to the query is to ignore the error.

If you want to surface the error, I'd recommend a few changes here:
1. Remove ~`ON CONFLICT DO NOTHING`~
2. Update the error handling:
    1. Identify the DB conflict error, like done [here](https://github.com/stellar/stellar-disbursement-platform-backend/blob/3663ea2228d72227669caba01e20e0d9871c7010/internal/data/url_shortener.go#L94-L98)
    2. Return a custom error, like this one:
    https://github.com/stellar/wallet-backend/blob/11a92dce4742c79220292a6d2db93b36e6ea8e36/internal/signing/store/keypairs_model.go#L16
    3.  In the Handler/mutationResolver, you should use `errors.Is` to identify if your custom error is in the stack, and then return a specific error response, instead of returning a `500`. I think a `409 - conflict` would be a good call here.

---

:wave: `RowsAffected` would probably be a good call if you want to return 404 when trying to delete a non-existing account. Then you'll need to apply error handling similar to what I described above for the insertion.